services/wireguard/consumer.go

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package wireguard
+
+import (
+	"errors"
+	"net"
+
+	"github.com/mdlayher/wireguardctrl/wgtypes"
+)
+
+var errNoIPAddress = errors.New("IP address required")
+
+type consumer struct {
+	allowedIPs net.IPNet
+	peer       wgtypes.PeerConfig
+	privateKey wgtypes.Key
+}
+
+// Config generates serviceConfiguration that will be used by Consumer to establish connection.
+func (c consumer) Config() (serviceConfig, error) {
+	if len(c.peer.AllowedIPs) == 0 {
+		return serviceConfig{}, errNoIPAddress
+	}
+
+	var config serviceConfig
+
+	config.Provider.Endpoint = *c.peer.Endpoint
+	config.Provider.PublicKey = c.peer.PublicKey
+	config.Consumer.PrivateKey = c.privateKey
+	config.AllowedIPs = c.allowedIPs
```

Comment:
From what I see this one will be initialised with '10.182.47.0/24', but is should be '0.0.0.0/0, ::/0', otherwise consumer will reach only local provider address. Or am I wrong?

Reply:
I don't think we need to setup allowed ips at all - we are not planning to do any ip based filtering at the moment - if we leave allowedip untouched nothing will break, and we can test floating ip feature (mobile app switches between different networks - but connection stays intact)

Reply:
Fixed.